src/cobra/summary/summary.py

```diff
@@ -0,0 +1,142 @@
+"""Provide the abstract base summary class."""
+
+
+import logging
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Optional, Union
+
+
+if TYPE_CHECKING:
+    from pandas import DataFrame
```

Comment:
Any particular reason why you are importing `pandas` inside the `TYPE_CHECKING` condition check?

Reply:
Not needed.

Reply:
Okay cool.